OpenRA.Game/World.cs

```diff
@@ -169,6 +169,8 @@ public bool ToggleInputMode<T>() where T : IOrderGenerator, new()
 			}
 		}
 
+		public readonly MusicPlaylist Playlist;
```

Comment:
In the `MediaGlobal` constructor add `playlist = world.WorldActor.Trait<MusicPlaylist>()` and a `readonly MusicPlaylist playlist;` instance variable.  You can do something similar in `MusicPlayerLogic`.